device/core/src/sak_signature_provider.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import {
+  SignatureProvider,
+  SharedAccessSignature,
+  errors
+} from 'azure-iot-common';
+
+export class SharedAccessKeySignatureProvider implements SignatureProvider {
+  private _tokenValidTimeInSeconds: number = 3600; // 1 hour
+
+  constructor(
+    private _sharedAccessKey: string,
+    tokenValidTimeInSeconds?: number
+  ) {
```

Comment:
nit: all our signatures are on the same single line in the rest of the code

Reply:
I used prettier to format all the new files I added. I can put it back into one line.

Reply:
are there rules to configure features of prettier?

Reply:
Yes, there's a `printWidth` setting that can be used to configure how long we want lines to be before it starts breaking them up.